src/ceph_osd.cc

```diff
@@ -628,12 +641,24 @@ int main(int argc, const char **argv)
     forker.exit(1);
   }
 
+  // check if the public_bind_addr option is set
+  if (!g_conf()->public_bind_addr.is_blank_ip()) {
+    bind_addrs = make_osd_addrs(g_conf()->public_bind_addr);
+  }
+
+  if (!g_conf()->public_addr.is_blank_ip()) {
+    public_addrs = make_osd_addrs(g_conf()->public_addr);
+  }
+
   if (ms_public->bindv(public_addrs) < 0)
```

Comment:
We can't assume that `public_addrs` in bindable if `public_bind_addr` is set. In such a scenario the former will be a purely virtual one IIRC.